src/platform/plugins/shared/workflows_management/public/features/debug-graph/workflow_graph_layout.ts

```diff
@@ -79,16 +80,16 @@ export function convertWorkflowGraphToReactFlow(graph: dagre.graphlib.Graph) {
 
   dagre.layout(dagreGraph);
 
-  const nodes = graph.nodes().map((id) => {
-    const dagreNode = dagreGraph.node(id);
-    const graphNode = graph.node(id) as any;
+  const nodes = graph.getAllNodes().map((graphNode) => {
+    const dagreNode = dagreGraph.node(graphNode.id);
+
     return {
-      id,
+      id: graphNode.id,
       data: {
         ...dagreNode,
         stepType: graphNode?.type,
-        step: graphNode?.configuration,
-        label: graphNode?.label || id,
+        step: (graphNode as any)?.configuration,
```

Comment:
why do we need casting here?

Reply:
not every node has configuration prop
since it's debug graph I did not care much